app/src/main/java/org/mifos/mobilebanking/presenters/LoginPresenter.java

```diff
@@ -75,7 +75,7 @@ public void detachView() {
      */
     public void login(final String username, final String password) {
         checkViewAttached();
-        if (isCredentialsValid(username, password)) {
+        if (isUsernameValid(username) & isPasswordValid(password)) {
```

Comment:
The build is failing because I'm using `&` here instead of `&&`. I cannot use **short circuit and** operation because both sides of the condition need to be called to show error on the username as well as the password field. @miPlodder As you said I'm not using any extra variables but this is failing on the **findbugs** check with the warning **NS_DANGEROUS_NON_SHORT_CIRCUIT: Potentially dangerous use of non-short-circuit logic**.